### DISCUSSION TITLE
issue: form is validated before pressing submit after submitting and resetting once before

### DISCUSSION THREAD

**Discussion by hornta** [react-hook-form/react-hook-form#8951] [Q&A]
### Version Number

7.34.2

### Codesandbox/Expo snack

https://codesandbox.io/s/quiet-hill-vdhucf?file=/src/App.tsx

### Steps to reproduce

1. Write something in input
2. Erase everything
3. Notice how no error message is shown
4. Write something and submit form
5. Write something in input
6. Erase everything
7. Notice how the error message is shown

### Expected behaviour

I don't expect to see an error message at step 7. I expect the form to be fully reset as it was when I landed on the page.

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by hornta** [comment]
I see this was forwarded to an discussion. Is this a bug or is the current way the expected behaviour?

**Comment by leapful** [comment]
@hornta : This is not a bug. You are currently resetting form data while submitting, therefore even with option "keepIsSubmitted", the form state for flag "isSubmitted" is sill set to "true" after all and trigger validation on change of the input.

A workaround for your use case is to perform "reset" async.

Working demo: https://codesandbox.io/s/upbeat-sky-hye6dc?file=/src/App.tsx

**Reply by hornta** [reply]
Alright thank you. It's not obvious. It makes me question if I do this correctly or if there are some other way of you know resetting a form upon submission like I do.
